src/lib.rs

```diff
@@ -52,11 +52,13 @@ pub fn desugar_book(
   book.generate_scott_adts();
   book.resolve_refs()?;
   encode_pattern_matching(book, &mut warnings)?;
+  book.check_unbound_vars()?;
   book.normalize_native_matches()?;
   book.check_unbound_vars()?;
   book.make_var_names_unique();
   book.linearize_vars();
   book.eta_reduction(eta);
+  book.check_unbounds()?;
```

Comment:
Same as above

Reply:
Also checking for unbound pats here doesn't make sense since they have already been encoded.
A sanity check of unbound patterns would be better after the intermediate steps where we're messing with the definitions (after extracting matches and after flatten, although i think it's not needed)